torchao/dtypes/uintx/gemlite_layout.py

```diff
@@ -173,6 +182,10 @@ def from_plain(
             exhaustive=False,
             use_cuda_graph=False,
         )
+        if _layout.group_size == None and _layout.bit_width == 4:
+                from gemlite.core import GEMLITE_ACC_DTYPE
+                from gemlite.dtypes import DType
+                GEMLITE_ACC_DTYPE[DType.FP16] = DType.FP32
```

Comment:
This will only work when all the layers use the same group_size, which is ok for now. 
The other option will be using this https://github.com/mobiusml/gemlite/blob/master/gemlite/core.py#L87 but for now let's keep it like this

Reply:
I tested this manually, it works in all cases even when there are different group sizes.

Reply:
I mean when different layers use different settings within the same model, but let's not worry about that !